homeassistant/components/eps/__init__.py

```diff
@@ -0,0 +1,88 @@
+"""EPS integration."""
+from __future__ import annotations
+
+import asyncio
+
+# from homeassistant.components.alarm_control_panel import SCAN_INTERVAL
+from datetime import timedelta
+import logging
+
+from pyepsalarm import EPS
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_TOKEN, CONF_USERNAME, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DATA_COORDINATOR, DOMAIN, EPS_TO_HASS
+
+PLATFORMS = [Platform.ALARM_CONTROL_PANEL]
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(seconds=60)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up EPS config."""
+    token: str = entry.data[CONF_TOKEN]
+    username: str = entry.data[CONF_USERNAME]
+    password: str = entry.data[CONF_PASSWORD]
+    eps_api = EPS(token, username, password)
+
+    # Use the EPS siteId as a unique_id
+    try:
+        async with asyncio.timeout(10):
+            site = await hass.async_add_executor_job(eps_api.get_site)
+    except (asyncio.TimeoutError, ConnectionError) as ex:
+        raise ConfigEntryNotReady from ex
+
+    coordinator = EPSDataUpdateCoordinator(hass, eps_api, site)
+    await coordinator.async_config_entry_first_refresh()
+
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][entry.entry_id] = {
+        DATA_COORDINATOR: coordinator,
+    }
```

Comment:
```suggestion
    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = coordinator
```

No need to overly complicate it and make it a dict when the only value is the coordinator. And you can combine those lines.